common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/SizeFinder.scala

```diff
@@ -1,29 +1,51 @@
 package uk.ac.wellcome.platform.archive.common.storage.services
 
 import com.amazonaws.services.s3.AmazonS3
-import uk.ac.wellcome.storage.ObjectLocation
+import com.amazonaws.services.s3.model.AmazonS3Exception
+import uk.ac.wellcome.storage.s3.S3Get
 import uk.ac.wellcome.storage.store.memory.MemoryStore
-
-import scala.util.Try
+import uk.ac.wellcome.storage.{
+  DoesNotExistError,
+  ObjectLocation,
+  ReadError,
+  StoreReadError
+}
 
 trait SizeFinder {
-  def getSize(location: ObjectLocation): Try[Long]
+  def getSize(location: ObjectLocation): Either[ReadError, Long]
 }
 
 class MemorySizeFinder(
   memoryStore: MemoryStore[ObjectLocation, Array[Byte]]
 ) extends SizeFinder {
-  override def getSize(location: ObjectLocation): Try[Long] = Try {
-    memoryStore.entries
-      .getOrElse(location, throw new Throwable(s"No such entry $location!"))
-      .length
-  }
+  override def getSize(location: ObjectLocation): Either[ReadError, Long] =
+    memoryStore.entries.get(location) match {
+      case Some(entry) => Right(entry.length)
+      case None =>
+        Left(DoesNotExistError(new Throwable(s"No such entry $location!")))
```

Comment:
I feel like there should be some way of identifying these errors as from the size finder, rather than recycling them from the storage service. Maybe compose a sizefinder trait in with all the errors you want? Or just copy paste?

Reply:
You reckon? I feel it's conceptually similar to what you'd get from a Readable instance: this is a `Readable[ObjectLocation, Long]`, so the storage errors seem to fit.